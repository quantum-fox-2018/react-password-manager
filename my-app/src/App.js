import React, { Component } from 'react';
import logo from './logo.svg';
import './App.css';
import {BrowserRouter as Router, Route, Switch} from 'react-router-dom'
import Home from './components/Home'
import Addpassword from './components/AddPassword'
import Editpassword from './components/EditPassword'
import LoginUser from './components/Login'
import SignUpUser from './components/Signup'
import PrivateRoute from './PrivateRoute'

class App extends Component {
  render() {
    return (
      <Router>
      <div className="App">
        <header className="App-header">
          <img src={logo} className="App-logo" alt="logo" />
          <h1 className="App-title">Password Manager</h1>
        </header>
        <Switch>
          <Route exact path ="/" component={LoginUser}/>
          <Route path="/signup" component={SignUpUser}/>
          <PrivateRoute path="/home" component={Home}/>
          <PrivateRoute path="/addpassword" component={Addpassword}/>
          <PrivateRoute path="/editpassword/:passId" component={Editpassword}/>
          <PrivateRoute path="*" 
            render = {() => (
              <h1>Page does not exist!</h1>
            )}
          />
        </Switch>
      </div>
      </Router>
    );
  }
}
// const userLogin = localStorage.getItem('token')
// const PrivateRoute = ({ component: Component, ...rest }) => (
//   <Route {...rest} render={(props) => (
//     userLogin ? <Component {...props} /> : <Redirect to='/' />
//   )} />
// )

export default App;
