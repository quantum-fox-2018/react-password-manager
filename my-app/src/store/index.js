import {createStore, applyMiddleware, combineReducers} from 'redux'
import thunk from 'redux-thunk'
import passwordReducers from './password/password.reducers'
import userReducers from './users/user.reducers'
import logger from 'redux-logger'


const reducers = combineReducers({
  passManager : passwordReducers,
  user: userReducers
})

const store = createStore(
  reducers,
  window.__REDUX_DEVTOOLS_EXTENSION__ && window.__REDUX_DEVTOOLS_EXTENSION__(),
  applyMiddleware(thunk, logger)
)

export default store