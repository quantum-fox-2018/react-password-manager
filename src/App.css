.App {
  text-align: center;
}

.App-logo {
  animation: App-logo-spin infinite 20s linear;
  height: 3rem;
  float: left;
}

.App-header {
  background-color: #222;
  height: 3rem;
  padding: 20px;
  color: white;
}

.App-title {
  font-size: 1.5em;
  float: left;
}

.App-title-right {
  background-color: #222;
  border: none;
  color: white;
  padding: 15px 32px;
  text-decoration: none;
  font-size: 16px;
  margin: 4px 2px;
  cursor: pointer;
  font-size: 1.5em;
  float: right;
}

.App-intro {
  font-size: large;
}

.container {
  margin: 2rem;
}

#showPass {
  border: 0px;
  background-color: transparent;
}

table {
  border-collapse: collapse;
  border-spacing: 0;
  width: 100%;
  border: 1px solid #ddd;
}

th, td {
  text-align: left;
  padding: 8px;
}

tr:nth-child(even){background-color: #f2f2f2}

input {
  width: 100%;
  padding: 12px 20px;
  margin: 8px 0;
  display: inline-block;
  border: 1px solid #ccc;
  border-radius: 4px;
  box-sizing: border-box;
}

.btn {
  width: 100%;
  background-color: #4CAF50;
  color: white;
  padding: 14px 20px;
  margin: 8px 0;
  border: none;
  border-radius: 4px;
  cursor: pointer;
  font-size: 18px;
}

span {
  font-size: 14px;
}

.btn:hover {
  background-color: #45a049;
}

.login {
  border-radius: 5px;
  background-color: #f2f2f2;
  padding: 20px;
}

@keyframes App-logo-spin {
  from { transform: rotate(0deg); }
  to { transform: rotate(360deg); }
}

#message {
  display:block;
  background: #f1f1f1;
  color: #000;
  position: relative;
  padding: 20px;
  margin-top: 10px;
}

#message p {
  padding: 10px 35px;
  font-size: 18px;
}

/* Add a green text color and a checkmark when the requirements are right */
.valid {
  color: green;
}

.valid:before {
  position: relative;
  left: -35px;
  content: "✔";
}

/* Add a red text color and an "x" when the requirements are wrong */
.invalid {
  color: red;
}

.invalid:before {
  position: relative;
  left: -35px;
  content: "✖";
}