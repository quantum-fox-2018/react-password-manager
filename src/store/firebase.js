import firebase from 'firebase'

const config = {
  apiKey: "AIzaSyCJ0Hlgag0DOOKpxzUsGTaE4FAu0tIE_EE",
  authDomain: "react-password-manager-bef30.firebaseapp.com",
  databaseURL: "https://react-password-manager-bef30.firebaseio.com",
  projectId: "react-password-manager-bef30",
  storageBucket: "react-password-manager-bef30.appspot.com",
  messagingSenderId: "827996324966"
}
firebase.initializeApp(config)
export const Database = firebase.database()
export const User = firebase.auth()