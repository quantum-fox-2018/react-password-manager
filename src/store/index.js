import { createStore, applyMiddleware, combineReducers } from "redux"
import logger from 'redux-logger'
import usersReducers from './users/reducers'
import thunk from 'redux-thunk'
import passmanagerReducer from "./passmanager/reducers";

const reducers = combineReducers({
  users: usersReducers,
  passmanager: passmanagerReducer
})

let store = createStore(reducers, window.__REDUX_DEVTOOLS_EXTENSION__ && window.__REDUX_DEVTOOLS_EXTENSION__(), applyMiddleware(thunk, logger))

export default store