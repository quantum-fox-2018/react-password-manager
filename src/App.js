import React, { Component } from 'react';
import logo from './logo.svg';
import './App.css';
import { BrowserRouter as Router, Route, Switch} from "react-router-dom";
import login from './components/login.jsx'
import register from './components/register.jsx'
import home from './components/home.jsx'

class App extends Component {
  logout = () => {
    if(window.confirm('Are you sure want to logout?')){
      localStorage.removeItem('token')
      window.location.reload()
    }
  }
  render() {
    return (
      <Router>
      <div className="App">
        <header className="App-header">
          <img src={logo} className="App-logo" alt="logo" />
          <h1 className="App-title">Password Manager</h1>
          <button className="App-title-right" onClick={this.logout}>Logout</button>
        </header>
        <Switch>
          <Route exact path="/" component={ login } />
          <Route path="/register" component={ register } />
          <Route path="/home" component={ home } />
        </Switch>
      </div>
      </Router>
    );
  }
}

export default App;
