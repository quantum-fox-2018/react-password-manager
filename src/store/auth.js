import { observable } from 'mobx'
import swal from 'sweetalert2'
import { auth } from './firebase'


class Auth {
  userId = ''
  email = ''
  @observable statusAuth = false

  errorNotif(err) {
    swal({
      position: 'center',
      type: 'error',
      title: 'Check your password',
      text: err.message
    })
  }

  successNotif(info) {
    swal({
      position: 'center',
      type: 'success',
      title: info,
      showConfirmButton: false,
      timer: 1000
    })
  }

  signUp(email, password) {
    auth.createUserWithEmailAndPassword(email, password)
    .then(res => {
      this.email = email
      this.userId = res.uid
      this.statusAuth = true
      this.successNotif('Enjoy!')
    })
    .catch(err => {
      this.errorNotif(err)
    })
  }

  signIn(email, password) {
    auth.signInWithEmailAndPassword(email, password)
    .then(res => {
      this.email = email
      this.userId = res.uid
      this.statusAuth = true
      this.successNotif('Welcome Back!')
    })
    .catch(err => {
      this.errorNotif(err)
    })
  }

  signOut() {
    this.email = ''
    this.userId = ''
    auth.signOut()
    .then(res => {
      this.successNotif('See You!')
      this.statusAuth = false
    })
    .catch(err => {
      this.errorNotif(err)
    })
  }
}

export default new Auth()
