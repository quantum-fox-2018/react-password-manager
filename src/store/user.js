import { observable } from 'mobx'
import swal from 'sweetalert2'


import { db, auth } from './firebase'
import authData from './auth'

class PassManager {
  @observable listData = []
  valUrl = false
  valLength = false
  valNumber = false
  valUpper = false
  valLower = false
  valSpecial = false
  allVal = false

  checkUrl(url) {
    let regex = /[-a-zA-Z0-9@:%_+.~#?&//=]{2,256}\.[a-z]{2,4}\b(\/[-a-zA-Z0-9@:%_+.~#?&//=]*)?/gi
    this.valUrl = regex.test(url)
    return this.valUrl
  }

  checkLength(password) {
    this.valLength = (password.length >= 6)
    return this.valLength
  }

  checkNumber(password) {
    this.valNumber = (/\d/g.test(password))
    return this.valNumber
  }

  checkUpper(password) {
    this.valUpper = (/[A-Z]/g.test(password))
    return this.valUpper
  }

  checkLower(password) {
    this.valLower = (/[a-z]/g.test(password))
    return this.valLower
  }

  checkSpecial(password) {
    this.valSpecial = (/\W/g.test(password))
    return this.valSpecial
  }

  checkAllValidation() {
    if(this.valUrl && this.valLength && this.valNumber && this.valUpper && this.valLower && this.valSpecial) {
      this.allVal = true
      return this.allVal
    }
  }

  hashPassword(str) {
    let length = str.length
    let bullet = String.fromCharCode(0x2022)
    return bullet.repeat(length)
  }

  showPassword(userId, idData) {
    swal.mixin({
      input: 'password',
      confirmButtonText: 'Confirm',
      showCancelButton: true,
    }).queue([
      {
        title: 'Password',
        text: 'enter your password'
      }
    ]).then((result) => {
      if (!result.value) {
        swal({
          type: 'question',
        })
      } else if (result.value[0]) {
        auth.signInWithEmailAndPassword(authData.email, result.value[0])
        .then(res => {
          let data = this.listData.map(data => {
            if(data.id === idData) {
              return {...data, showPassword: !data.showPassword}
            } else {
              return {...data}
            }
          })
          this.listData = data  
        })
        .catch(err => {
          swal({
            type: 'error',
            title: 'Error',
            text: err.message
          })
        })
      }
    })
  }

  loadData(userId) {
    db.ref(`/user/${userId}`).once('value', (snap) => {
      const pass = snap.val()
      let arr = []
      for (const key in pass) {
        if (pass.hasOwnProperty(key)) {
          const el = pass[key]
          const update = {...el, id: key, showPassword: false}
          arr.push(update)
        }
      }
      this.listData = arr
    })
  }

  createData(userId, url, username, password) {
    db.ref(`/user/${userId}`).push({
      url, username, password, createdAt: Date.now(), updatedAt: null
    })
    .then(res => {
      swal({
        position: 'center',
        type: 'success',
        title: 'Your password has been saved',
        showConfirmButton: false,
        timer: 1500
      })
      this.loadData(userId)
    })
    .catch(err => {
      alert(err)
    })
  }

  deleteData(userId, idData) {
    swal({
      title: 'Are you sure?',
      text: "You won't be able to revert this!",
      type: 'warning',
      showCancelButton: true,
      confirmButtonColor: '#3085d6',
      cancelButtonColor: '#d33',
      confirmButtonText: 'Yes, delete it!'
    }).then((result) => {
      if (result.value) {
        db.ref(`/user/${userId}`).child(idData).remove()
        this.loadData(userId)
        swal(
          'Deleted!',
          'Your file has been deleted.',
          'success'
        )
      }
    })
  }

  editPassword(userId, idData) {
    swal.mixin({
      input: 'password',
      confirmButtonText: 'Confirm',
      showCancelButton: true,
      inputAttributes: {
        'minlength': 6,
        'autocapitalize': 'off',
        'autocorrect': 'off'
      }
    }).queue([
      {
        title: 'Edit Password',
        text: 'enter your new password'
      }
    ]).then((result) => {
      if (result.value) {
        let data = this.listData.map(data => {
          if(data.id === idData) {
            let newData = {...data, password: result.value[0], updatedAt: Date.now()}
            db.ref(`/user/${userId}`).child(idData).update(newData)
            return newData
          } else {
            return {...data}
          }
        })
        this.listData = data
        swal({
          position: 'center',
          type: 'success',
          title: 'Your password has been edited',
          showConfirmButton: false,
          timer: 1500
        })
      } else {
        swal({
          type: 'question',
        })
      }
    })
  }
}

export default new PassManager()
