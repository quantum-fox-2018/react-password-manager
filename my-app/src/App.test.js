import React from 'react';
import ReactDOM from 'react-dom';
import Enzyme, {shallow, mount} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import {Provider} from 'react-redux'
import {Route, Switch} from 'react-router-dom'
import store from './store'
import Privateroute from './test/localstorageMock'
import renderer from 'react-test-renderer'

global.localStorage = new LocalStorageMock();
// let localStorage = new LocalStorageMock();
Enzyme.configure({ adapter: new Adapter() });

import LocalStorageMock from './test/localstorageMock'
import App from './App';


describe('<App/>', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    const wrapper = mount(
    <Provider store={store}>
    <App/>
    </Provider>
    )
    ReactDOM.render(wrapper, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('should render <Route/> <Privateroute/> <Switch/>', () => {
    const wrapper = shallow(<App/>)
    expect(wrapper.containsAllMatchingElements([
      <Switch/>,
      <Route/>,
      <Privateroute/>,
    ])).toHaveLength[7]
  })

  it('renders without crashing', () => {
    const rendered = renderer.create(
    <Provider store={store}>
      <App/>
    </Provider>)
    expect(rendered).toMatchSnapshot()
  });
})


