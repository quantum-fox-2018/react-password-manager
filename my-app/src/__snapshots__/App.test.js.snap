// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App/> renders without crashing 1`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    <img
      alt="logo"
      className="App-logo"
      src="logo.svg"
    />
    <h1
      className="App-title"
    >
      Password Manager
    </h1>
  </header>
  <div
    className="container"
  >
    <form>
      <div
        className="form-group"
      >
        <label>
          Username
        </label>
        <input
          className="form-control"
          id="username"
          name="username"
          onChange={[Function]}
          type="text"
          value=""
        />
      </div>
      <div
        className="form-group"
      >
        <label>
          Password
        </label>
        <input
          className="form-control"
          id="password"
          name="password"
          onChange={[Function]}
          type="password"
          value=""
        />
      </div>
      <button
        className="btn btn-primary"
        onClick={[Function]}
        type="button"
      >
        Login
      </button>
    </form>
    <div
      className="container"
    >
      <p>
        New user?
      </p>
      <a
        href="/signup"
        onClick={[Function]}
      >
        Sign up Here
      </a>
    </div>
  </div>
</div>
`;
